foundation/plugins/org.jboss.tools.foundation.core/src/org/jboss/tools/foundation/core/credentials/internal/CredentialDomain.java

```diff
@@ -59,8 +65,20 @@ public CredentialDomain(Preferences pref) throws BackingStoreException {
 				credentials.put(users[i], NOT_LOADED_PASSWORD);
 			}
 		}
-		if( defaultUsername == null && credentials.size() > 0 ) {
-			defaultUsername = credentials.keySet().iterator().next();
+		
+		String promptedUserList = pref.get(PROPERTY_PROMPTED_USER_LIST, (String)null);
+		promptedCredentials = new ArrayList<String>();
+		if( !promptedUserList.isEmpty()) {
+			String[] users = (usersList == null ? new String[0] : promptedUserList.split("\n"));
+			promptedCredentials.addAll(Arrays.asList(users));
+		}
+		
+		if( defaultUsername == null || !userExists(defaultUsername)) {
+			// The default name doesn't exist, so we need another. 
+			String[] users = getUsernames();
+			if( users.length > 0) {
```

Comment:
This situation will probably only be reached if the user has just removed the default username. For example, userlist = { "rob", "fred"}  where "rob" is the current already-set default. If the user then deletes this username, the next one in the list becomes the default. The user can still change this by clicking on the domain and selecting 'edit' where he can select which of the available usernames is the default.  

So while I agree, I think this is the best compromise via UI. It'd be very weird for the user to delete the current default username and then have a popup or some other UI element forcing him to choose the next default. It's more logical for the next in the list to become the default, and if the user has a problem with it, he can set it by editing the domain.